xilem_web/src/interfaces.rs

```diff
@@ -204,449 +232,2100 @@ where
+//     HtmlElement<State, Action, DomNode: AsRef<web_sys::HtmlParamElement>>
+// {
+// }
+
+// #[cfg(feature = "HtmlParamElement")]
+// impl<State, Action, T> HtmlParamElement<State, Action> for T
+// where
+//     T: HtmlElement<State, Action>,
+//     T::DomNode: AsRef<web_sys::HtmlParamElement>,
+// {
+// }
+
+// #[cfg(feature = "HtmlPictureElement")]
+pub trait HtmlPictureElement<State, Action = ()>:
+    HtmlElement<State, Action, DomNode: AsRef<web_sys::HtmlPictureElement>>
+{
+}
+
+// #[cfg(feature = "HtmlPictureElement")]
+impl<State, Action, T> HtmlPictureElement<State, Action> for T
+where
+    T: HtmlElement<State, Action>,
+    T::DomNode: AsRef<web_sys::HtmlPictureElement>,
+{
+}
+
+// #[cfg(feature = "HtmlPreElement")]
+pub trait HtmlPreElement<State, Action = ()>:
+    HtmlElement<State, Action, DomNode: AsRef<web_sys::HtmlPreElement>>
+{
+}
+
+// #[cfg(feature = "HtmlPreElement")]
+impl<State, Action, T> HtmlPreElement<State, Action> for T
+where
+    T: HtmlElement<State, Action>,
+    T::DomNode: AsRef<web_sys::HtmlPreElement>,
+{
+}
+
+// #[cfg(feature = "HtmlProgressElement")]
+pub trait HtmlProgressElement<State, Action = ()>:
+    HtmlElement<State, Action, DomNode: AsRef<web_sys::HtmlProgressElement>>
+{
+}
+
+// #[cfg(feature = "HtmlProgressElement")]
+impl<State, Action, T> HtmlProgressElement<State, Action> for T
+where
+    T: HtmlElement<State, Action>,
+    T::DomNode: AsRef<web_sys::HtmlProgressElement>,
+{
+}
+
+// #[cfg(feature = "HtmlQuoteElement")]
+pub trait HtmlQuoteElement<State, Action = ()>:
+    HtmlElement<State, Action, DomNode: AsRef<web_sys::HtmlQuoteElement>>
+{
+}
+
+// #[cfg(feature = "HtmlQuoteElement")]
+impl<State, Action, T> HtmlQuoteElement<State, Action> for T
+where
+    T: HtmlElement<State, Action>,
+    T::DomNode: AsRef<web_sys::HtmlQuoteElement>,
+{
+}
+
+// #[cfg(feature = "HtmlScriptElement")]
+pub trait HtmlScriptElement<State, Action = ()>:
+    HtmlElement<State, Action, DomNode: AsRef<web_sys::HtmlScriptElement>>
+{
+}
+
+// #[cfg(feature = "HtmlScriptElement")]
+impl<State, Action, T> HtmlScriptElement<State, Action> for T
+where
+    T: HtmlElement<State, Action>,
+    T::DomNode: AsRef<web_sys::HtmlScriptElement>,
+{
+}
+
+// #[cfg(feature = "HtmlSelectElement")]
+pub trait HtmlSelectElement<State, Action = ()>:
+    HtmlElement<State, Action, DomNode: AsRef<web_sys::HtmlSelectElement>>
+{
+    /// A string representing the value of the HTMLOptionElement, i.e. the value attribute of the equivalent `<option>`.
+    /// If this is not specified, the value of text is used as the value, e.g. for the associated `<select>` element's value when the form is submitted to the server.
+    ///
+    /// See <https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/value> for more details
+    fn value(self, value: impl IntoAttributeValue) -> Attr<Self, State, Action> {
+        Attr::new(self, Cow::from("value"), value.into_attr_value())
+    }
```

Comment:
My vision of the API is obviously builder/modifier-pattern like as seen with all these DOM interface extension traits, and the examples.
So I added all these attributes (seen in the `blog` example, which doesn't use any `Element::attr`) as methods, currently just sugaring the `Attr` view.
These are of course not complete, I only added the attributes needed in the `blog` example, and added MDN docs on top of some.
I expect this to be extended over time (together with the feature gates explained above).